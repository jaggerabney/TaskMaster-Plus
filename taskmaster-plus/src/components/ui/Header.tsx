"use client";

import Button from "./Button";

export default function Header() {
  return (
    <nav className="w-full bg-redNCS p-8 flex flex-row justify-between">
      <h1
        id="header-title"
        className="text-4xl select-none font-bold text-white"
      >
        TaskMaster+
      </h1>
      <div className="w-1/12">
        <Button
          id="header-login-button"
          text="Login"
          textColor="text-white"
          borderColor="border-white"
          onClick={() => {}}
        />
      </div>
    </nav>
  );
}
