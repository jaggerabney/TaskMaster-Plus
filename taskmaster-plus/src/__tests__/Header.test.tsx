import "@testing-library/jest-dom";
import { render } from "@testing-library/react";

import Header from "@/components/ui/Header";

describe("Header component", () => {
  it("renders correctly", () => {
    const header = render(<Header />);
    expect(header).toMatchSnapshot();
  });

  it("title component is not selectable", () => {
    render(<Header />);
    const title = document.getElementById("header-title");
    expect(title).toHaveClass("select-none");
  });

  it("button does nothing (for now!)", () => {
    render(<Header />);
    const loginButton = document.getElementById("header-login-button");
    expect(loginButton).toBeEnabled(); // will be changed when auth logic is added
  });
});
