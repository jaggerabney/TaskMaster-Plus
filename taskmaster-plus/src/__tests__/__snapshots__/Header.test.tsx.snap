// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="w-full bg-redNCS p-8 flex flex-row justify-between"
      >
        <h1
          class="text-4xl select-none font-bold text-white"
          id="header-title"
        >
          TaskMaster+
        </h1>
        <div
          class="w-1/12"
        >
          <button
            class="px-4 py-2 w-full text-white border-white undefined whitespace-nowrap select-none font-bold border-2 rounded hover:scale-[1.03] duration-100"
            id="header-login-button"
          >
            Login
          </button>
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="w-full bg-redNCS p-8 flex flex-row justify-between"
    >
      <h1
        class="text-4xl select-none font-bold text-white"
        id="header-title"
      >
        TaskMaster+
      </h1>
      <div
        class="w-1/12"
      >
        <button
          class="px-4 py-2 w-full text-white border-white undefined whitespace-nowrap select-none font-bold border-2 rounded hover:scale-[1.03] duration-100"
          id="header-login-button"
        >
          Login
        </button>
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
